### DISCUSSION TITLE
PIP-17: PNS Pactus Name System

### DISCUSSION THREAD

**Discussion by ja7ad** [pactus-project/PIPs#75] [PIPs]
PIP-76: [PNS Pactus Name System](https://pips.pactus.org/PIPs/pip-17)

**Comment by b00f** [comment]
@kehiy  It is really good idea to define PNS (or name service) as ETh did.
However, we don't support smart contract now. Specially the smart contract APIs are not defined yet.
Here, this proposal, assumes that this smart contract has a "main" function named "register".
I am not sure if the final smart contract in Pactus follow this pattern.

**Reply by kehiy** [reply]
@b00f 

The base idea behind PNS is an Intenet-like DNS over the blockchain. smart contract patterns can be updated in the future.

**Reply by b00f** [reply]
Agree, but before that we need a solution for storage to manage and store the Names.

**Reply by kehiy** [reply]
Based on PIP-17 itself, the way that nodes save, cache, and serve them is up to node implementations. 
They MUST detect these transactions, save them somewhere, and serve last updates for registrars in their APIs. the way they do that is up to them.
